rts/motoko-rts/src/gc/mark_compact/bitmap.rs

```diff
@@ -22,19 +23,29 @@ pub unsafe fn alloc_bitmap<M: Memory>(mem: &mut M, heap_size: Bytes<u32>) {
 
 pub unsafe fn free_bitmap() {
     BITMAP_PTR = core::ptr::null_mut();
+    BITMAP_BASE = 0
+}
+
+/// Move the bitmap base such that accessing `idx`
+/// will touch the beginning of the bitmap.
+pub unsafe fn translate_bitmap(idx: u32) {
+    let (byte_idx, bit_idx) = (idx / 8, idx % 8);
+    assert_eq!(bit_idx, 0);
+    BITMAP_BASE = byte_idx as usize;
+    BITMAP_PTR = BITMAP_PTR.sub(BITMAP_BASE);
```

Comment:
I'm confused about this code. `BITMAP_PTR` after allocating the bitmap points to a blob payload. Now you subtract a few bytes from it. How do you make sure this won't overwrite the blob header, or dynamic objects allocated before the blob?

Reply:
I think I understand this now, see my top-level comment.

Reply:
I have integrated this into the `alloc_bitmap` code.